File: frontend/src/types/transaction.ts
/**
 * 交易相关类型定义
 */

import type { TransactionType } from './common';

export type Module = 'dividend' | 'fixed' | 'growth' | 'allweather';

export interface Transaction {
  id: number;
  positionId: number | null;
  symbol: string;
  name: string | null;
  module: Module | null;
  transactionType: TransactionType;
  shares: number;
  price: number;
  totalAmount: number;
  fee: number;
  currency: string;
  transactionDate: string;
  notes: string | null;
  realizedProfit: number | null;
  createTime: string;
}

export interface TransactionRequest {
  positionId?: number;
  symbol: string;
  transactionType: TransactionType;
  shares: number;
  price: number;
  fee?: number;
  currency?: string;
  transactionDate: string;
  notes?: string;
}

export interface TransactionQuery {
  pageNum?: number;
  pageSize?: number;
  transactionType?: TransactionType | '';
  module?: Module | '';
  keyword?: string;
  startDate?: string;
  endDate?: string;
}

export interface TransactionSummary {
  totalCount: number;
  buyCount: number;
  sellCount: number;
  buyAmount: number;
  sellAmount: number;
  totalFee: number;
  realizedProfit: number;
  realizedProfitRate: number;
  moduleSummaries: Array<{
    module: Module;
    moduleName: string;
    count: number;
    amount: number;
  }>;
}

File: backend/asset-domain/src/main/java/com/asset/domain/entity/Transaction.java
package com.asset.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 交易记录实体
 */
@Data
@TableName("transactions")
public class Transaction {

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 持仓ID
     */
    private Long positionId;

    /**
     * 标的代码
     */
    private String symbol;

    /**
     * 标的名称
     */
    private String name;

    /**
     * 所属模块（dividend/fixed/growth/allweather）
     */
    private String module;

    /**
     * 市场类型（a_stock/etf/fund/hk_stock/us_stock）
     */
    private String market;

    /**
     * 交易类型
     */
    private String transactionType;

    /**
     * 交易数量
     */
    private BigDecimal shares;

    /**
     * 成交价格
     */
    private BigDecimal price;

    /**
     * 成交金额
     */
    private BigDecimal totalAmount;

    /**
     * 手续费
     */
    private BigDecimal fee;

    /**
     * 币种
     */
    private String currency;

    /**
     * 交易日期
     */
    private LocalDate transactionDate;

    /**
     * 备注
     */
    private String notes;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 创建人
     */
    private String createBy;

    /**
     * 逻辑删除标识
     */
    private Integer deleted;
}